### PR TITLE
Add natural sorting (sorting case insensitively as well as numerically)

### PR DESCRIPTION
**Pull request by lancestarr**
This pull request adds a new option for sorting things like tab completion results.

It sorts a list like this: ['a1', 'A2', 'a3', 'A11', 'a22'].

### CODE REVIEW

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -1093,8 +1093,8 @@ def complete_users():
             self.allow_appended_space = False
             self.allow_closing_quote = False
 
-        # Sort the matches before any trailing slashes are added
-        matches = utils.alphabetical_sort(matches)
+        # Sort the matches alphabetically before any trailing slashes are added
+        matches.sort(key=utils.norm_fold)
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: Nice little performance improvement to prevent making an extra copy

File: cmd2/utils.py [hunk 2]
```diff
@@ -172,7 +173,52 @@ def norm_fold(astr: str) -> str:
 def alphabetical_sort(list_to_sort: List[str]) -> List[str]:
     """Sorts a list of strings alphabetically.
 
+    For example: ['a1', 'A11', 'A2', 'a22', 'a3']
```
Comment (tleonhardt) [inline cmd2/utils.py]: Thank you for creating good docstrings

File: tests/test_utils.py [hunk 3]
```diff
@@ -48,5 +48,31 @@ def test_unicode_casefold():
...
+def test_natural_keys():
+    my_list = ['café', 'µ', 'A', 'micro', 'unity', 'x1', 'X2', 'X11', 'X0', 'x22']
+    my_list.sort(key=cu.natural_keys)
+    assert my_list == ['A', 'café', 'micro', 'unity', 'X0', 'x1', 'X2', 'X11', 'x22', 'µ']
+    my_list = ['a3', 'a22', 'A2', 'A11', 'a1']
+    my_list.sort(key=cu.natural_keys)
+    assert my_list == ['a1', 'A2', 'a3', 'A11', 'a22']
+
+def test_natural_sort():
+    my_list = ['café', 'µ', 'A', 'micro', 'unity', 'x1', 'X2', 'X11', 'X0', 'x22']
```
Comment (tleonhardt) [inline tests/test_utils.py]: Nice unit tests - they do a good job of communicating intended behavior

**Comment by tleonhardt** [comment]
@lancestarr Thanks for the PR Lance!  I'm sure more than one end user would benefit from a built-in natural sort (as opposed to alphabetical sort).

File: cmd2/cmd2.py (hunk 1 above)
Comment (lancestarr) [reply cmd2/cmd2.py]: Thanks!

File: cmd2/utils.py (hunk 2 above)
Comment (lancestarr) [reply cmd2/utils.py]: Thanks! Kevin made me do it.

**Pull request merged**